src/js/renderers/ArrayLayoutRenderer.js

```diff
@@ -1,223 +1,207 @@
-  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-  THE SOFTWARE.
-*/
 import range from "lodash/range"
-import React, { useMemo, useContext, useEffect } from 'react';
+import React, { useMemo } from 'react';
 import {
-  composePaths,
-  createDefaultValue,
-  findUISchema,
-  Helpers,
-  rankWith,
-  or,
-  isPrimitiveArrayControl,
-  isObjectArrayControl,
-  isObjectArrayWithNesting
+	composePaths,
+	createDefaultValue,
+	findUISchema,
+	Helpers
 } from "@jsonforms/core"
 import {
   JsonFormsDispatch,
   withJsonFormsArrayControlProps
 } from "@jsonforms/react"
 
-import { Context as NavigatorContext } from '../component/context';
-
 import { chevronLeft, chevronRight, plus } from '@wordpress/icons';
 import { isRTL, __ } from '@wordpress/i18n';
 import { IconWithCurrentColor } from './NavigatorLayout/icon-with-current-color';
 import { NavigationButtonAsItem } from './NavigatorLayout/navigation-button';
 
 import {
+	__experimentalUseNavigator as useNavigator,
 	__experimentalHStack as HStack,
     __experimentalItemGroup as ItemGroup,
 	__experimentalItem as Item,
 	FlexItem,
+	Button
 } from '@wordpress/components';
 
 const { convertToValidClassName } = Helpers
 
-export const ArrayControl = ({
-  classNames,
-  data,
-  label,
-  path,
-  schema,
-  errors,
-  addItem,
-  removeItems,
-  moveUp,
-  moveDown,
-  uischema,
-  uischemas,
-  getStyleAsClassName = (cls) => cls,
-  renderers,
-  rootSchema,
-  translations
-}) => {
-  const controlElement = uischema
-  const childUiSchema = useMemo(
-    () =>
-      findUISchema(
-        uischemas,
-        schema,
-        uischema.scope,
-        path,
-        undefined,
-        uischema,
-        rootSchema
-      ),
-    [uischemas, schema, uischema.scope, path, uischema, rootSchema]
-  )
-  const isValid = errors.length === 0
-  const validationClass = getStyleAsClassName("array.control.validation")
-  const divClassNames = [validationClass]
-    .concat(
-      isValid ? "" : getStyleAsClassName("array.control.validation.error")
-    )
-    .join(" ")
-  const buttonClassAdd = getStyleAsClassName("array.control.add")
-  const labelClass = getStyleAsClassName("array.control.label")
-  const childControlsClass = getStyleAsClassName("array.child.controls")
-  const buttonClassUp = getStyleAsClassName("array.child.controls.up")
-  const buttonClassDown = getStyleAsClassName("array.child.controls.down")
-  const buttonClassDelete = getStyleAsClassName("array.child.controls.delete")
-  const controlClass = [
-    getStyleAsClassName("array.control"),
-    convertToValidClassName(controlElement.scope)
-  ].join(" ")
+// TODO: add new item button component
+const AddItemButton = ({ route, path, label, schema, addItem }) => {
+	const navigator = useNavigator();
+	return (
+		<Button
+			aria-label={ `Add new item` }
+			onClick={ () => {
+				addItem( path, createDefaultValue(schema) )();
```

Comment:
Why call the addItem()? Because it [returns dispatch function](https://github.com/eclipsesource/jsonforms/blob/d6f1fda00cdf9502c144b3706ebc7f555a8c6fee/packages/core/src/util/renderer.ts#L786)